### ISSUE TITLE
Re-running terraform apply after initial Error will error out with "domain [...] already exists"

### ISSUE THREAD

**Issue by raoulbhatia** [dmacvicar/terraform-provider-libvirt#921]
Related to #920 and also #423

##  System Information etc.

See #920
___

### Setup

See #920 

### Steps to Reproduce Issue

1. `terraform apply` until you hit #920 
2. Manually fix by editing `/etc/apparmor.d/local/abstractions/libvirt-qemu`
3. `terraform apply`

```
libvirt_domain.domain-ubuntu: Creating...
╷
│ Error: error defining libvirt domain: operation failed: domain 'terraform-kvm-ansible' already exists with uuid 7454585d-3a7c-42a9-a822-835b9ca3c1b6
│
│   with libvirt_domain.domain-ubuntu,
│   on main.tf line 33, in resource "libvirt_domain" "domain-ubuntu":
│   33: resource "libvirt_domain" "domain-ubuntu" {
│
```

### Solution

1. Run `virsh undefine terraform-kvm-ansible`
2. `terraform apply`
3. `terraform-kvm-ansible` will come up successfully

___
## Additional information:

Apparmor as articulated in #920

**Comment by alancaldelas** [comment]
This also is happening to me, any updates?

**Comment by tovoro** [comment]
Same here while deploying Centos 8 guests on a Fedora35 Host

**Comment by alancaldelas** [comment]
Yeah, I'm also on a F35 host. Additionally, I'm doing a UEFI deployment and running the destroy also removes the /usr/share/edk2/OVMF_VARS.fd file so when I got re-deploy the file is now missing.

**Comment by scabala** [comment]
Hi,
could you try with latest provider version?

**Comment by radarsat1** [comment]
I've been trying to do a very basic set up of 2 VMs as a first try for this provider, and it just fails when trying to create the domains:

```
libvirt_cloudinit_disk.commoninit: Creating...
libvirt_volume.k8s-cloudinit[0]: Creating...
libvirt_volume.k8s-cloudinit[1]: Creating...
libvirt_volume.k8s-cloudinit[1]: Creation complete after 0s [id=/var/lib/libvirt/images/worker]
libvirt_volume.k8s-cloudinit[0]: Creation complete after 1s [id=/var/lib/libvirt/images/master]
libvirt_cloudinit_disk.commoninit: Creation complete after 1s [id=/var/lib/libvirt/images/commoninit.iso;3637b962-9332-4963-adb2-ff1b13998806]
libvirt_domain.domain-k8s[1]: Creating...
libvirt_domain.domain-k8s[0]: Creating...
╷
│ Error: error defining libvirt domain: operation failed: domain 'worker' already exists with uuid 19cb3416-c22c-41d7-a752-3d2d828ed359
│ 
│   with libvirt_domain.domain-k8s[1],
│   on main.tf line 39, in resource "libvirt_domain" "domain-k8s":
│   39: resource "libvirt_domain" "domain-k8s" {
│ 
╵
╷
│ Error: error defining libvirt domain: operation failed: domain 'master' already exists with uuid ace01876-d283-4f7a-af35-b0a5d10af182
│ 
│   with libvirt_domain.domain-k8s[0],
│   on main.tf line 39, in resource "libvirt_domain" "domain-k8s":
│   39: resource "libvirt_domain" "domain-k8s" {
│ 
╵
```

but nothing is listed,

```
$ virsh list --all
 Id   Name   State
--------------------
```

and I can't undefine them,

```
$ virsh undefine master
error: failed to get domain 'master'


$ virsh undefine worker
error: failed to get domain 'worker'

```

So I have to conclude that this provider is just broken.  Although I have no idea where it's getting the uuids from if they are not listed with virsh.  I just fresh installed with the latest version so it seems to be using v0.8.3.

**Comment by dmacvicar** [comment]
ℹ️ ℹ️ ℹ️ ℹ️ ℹ️ ℹ️

This issue report is relevant to the legacy version of the provider, which is now in the v0.8 branch of this repository.

Future development is based on a new provider, which is not compatible with this one, nor does share code.

As the new provider solves many issues with the legacy, and to make development in my free time more enjoyable, I have decided to close all bugs for the legacy provider, and to ask to check if the bug would apply to the new one. This also to encourage trying the new version.

- https://registry.terraform.io/providers/dmacvicar/libvirt/latest

and check the documentation:

- https://github.com/dmacvicar/terraform-provider-libvirt/blob/main/README.md
- https://registry.terraform.io/providers/dmacvicar/libvirt/latest/docs

You are free to reopen the bug for v0.8. It may encourage someone to fix it. My efforts will go into the new provider.

I ask you to check the new provider and re-evaluate this bug. 🙏
